Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -207,45 +212,44 @@ import BraintreeCore
             apiClient.post("", parameters: graphQLParameters, httpType: .graphQLAPI) { body, _, error in
                 if let error = error as? NSError {
                     if error.code == BTCoreConstants.networkConnectionLostCode {
-                        self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.network-connection.failure")
+                        self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeNetworkConnectionLost)
                     }
 
-                    self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.client-failure")
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                     self.appSwitchCompletion(nil, error)
                     return
                 }
 
                 guard let body else {
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                     self.appSwitchCompletion(nil, BTVenmoError.invalidBodyReturned)
                     return
                 }
 
                 let venmoAccountNonce: BTVenmoAccountNonce = BTVenmoAccountNonce(with: body)
-                self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.success")
 
                 if self.shouldVault && self.apiClient.clientToken != nil {
                     self.vault(venmoAccountNonce.nonce)
                 } else {
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeSucceeded)
                     self.appSwitchCompletion(venmoAccountNonce, nil)
                     return
                 }
             }
 
         case .succeeded:
             guard let nonce = returnURL.nonce else {
-                apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.client-failure")
+                apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                 appSwitchCompletion(nil, BTVenmoError.invalidReturnURL("nonce"))
                 return
             }
 
             guard let username = returnURL.username else {
-                apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.client-failure")
+                apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                 appSwitchCompletion(nil, BTVenmoError.invalidReturnURL("username"))
                 return
             }
 
-            apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.success")
-
             if shouldVault && apiClient.clientToken != nil {
                 vault(nonce)
```

Comment:
I think we probably need to add the tokenize events to the `vault` method since if we hit this block we'd never end up sending a success/failure.

Reply:
I was wondering about the second else block in the case the clientToken is nil. That would be a fail, right?

Reply:
Ah, I think I must have missed those. 👍

Reply:
I am only seeing checks that `clientToken != nil` for vaulting. It looks like we check for the existence of a client token to vaultVenmo and if we don't vault we continue on the non-vault flow. If I misunderstood the conditional you were talking about feel free to link the code block you were referencing here!

Reply:
yeah, that's what i was thinking. We need to add the case clientToken is nil on top of both of these? wdyt?

Reply:
@scannillo is this logic checking for client token not being nil only for vault option ok as it is?